### PR TITLE
Closes #2 Implements initModel and tests it

### LINKED ISSUE #2: Implement init model
+#Model structure:
+#list of:
+# - best point globally known
+#     - coordinates
+#     - quality
+# - list of points
+#     - vector of best known coordinates for point
+#     - vector of current coordinates
+#     - vector of velocity
+#     - quality

### PR DESCRIPTION
**Pull request by karolmie1**
Function normalizeBySquaring() won't work in all situations, but fixing it seems outside of scope, so I have created issue to fix that in future.

Sample usage of model is can be found in tests.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-How to run it:
+aszdfasdf:How to run it:
```
Comment (mklecha) [inline readme.md]: To be deleted

File: metaheuristic_pattern.r [hunk 2]
```diff
@@ -22,6 +22,20 @@ getBestPoint <- function(points) {
...
 }
 
+#TODO: this is biased by distance from zero, it should be biased by the middle point between extremes: implementation and tests
+normalizeBySquaring <- function(velocity, center) {
+  return(sign(velocity)*sqrt(abs(velocity)));
+}
+
+#Initial Velocity should be random, and should be adequate to search space size
+#This function provides such constrained initial velocities.
+generateInitVelocity <- function(popCount, dimCount) {
```
Comment (mklecha) [inline metaheuristic_pattern.r]: Why "pop" in popCount? To me this name is meaningless and unclear

File: metaheuristic_pattern.r [hunk 3]
```diff
@@ -33,9 +47,11 @@ getBestPoint <- function(points) {
 #     - vector of velocity
 #     - quality
 initModel<-function(history) {
-#TODO: finish to comply with model
-  newModel = list(point=history, best=getBestPoint(history));
-   return(newModel);
+  popCount = length(history);
+  dimCount = length(history[[1]]$coordinates);
+  population = list(velocity = generateInitVelocity(popCount, dimCount), position = history, best = history);
```
Comment (mklecha) [inline metaheuristic_pattern.r]: I believe if we have array of velocities we should call it velocities, and so on...

File: metaheuristic_pattern.r (hunk 2 above)
Comment (karolmie1) [inline metaheuristic_pattern.r]: Point taken. For me it was shortcut for populationCount, but i guess we particles would be better.

**Pull request merged**